Networking Demo AppTests/APIClientTests.swift

```diff
@@ -77,5 +77,41 @@ class APIClientTests: XCTestCase {
         }
         waitForExpectations(timeout: 10, handler: nil)
     }
+    
+    @available(iOS 15.0.0, *)
+    func testAsyncValidCreateRequest() async throws {
+        let apiClientExpectation = expectation(description: "expect apiClient call back to be called.")
+        
+        let name = UUID().uuidString
+        let newUser = UserModel(id: Int.random(in: 0...1000),
+                                name: name,
+                                email: "\(name)@gmail.com",
+                                gender: .female,
+                                status: .active)
+        let validCreateRequest = CreateUsersRequest(userModel: newUser)
+        switch await apiClient.perform(validCreateRequest) {
+        case .success:
+            apiClientExpectation.fulfill()
+        case .failure:
+            XCTFail()
+        }
+        
+        await waitForExpectations(timeout: 10, handler: nil)
+    }
+    
+    @available(iOS 15.0.0, *)
+    func testAsyncInvalidCreateRequest() async throws {
+        let apiClientExpectation = expectation(description: "expect apiClient call back to be called.")
```

Comment:
technically we don't, but i wanted to assert at least something to check that the test works as intended in line 94. otherwise it would be just `break`

Reply:
here you already have an assertion, we don't need an expectation.